docs/references/astro/user-store.mdx

```diff
@@ -0,0 +1,115 @@
+---
+title: $userStore
+description: The $userStore store is used to get the current user object and to update the user's data on the client side.
+---
+
+# `$userStore`
+
+The `$userStore` store is a convenient way to access the current [`User`](/docs/references/javascript/user/user) data where you need it. This store provides the user data and helper methods to manage the current active session.
+
+## How to use the `$userStore` store
+
+### Retrieve the current user data with the `$userStore` store
+
+The following example demonstrates how to use the `$userStore` store to access the `user` object, which includes the current user's data, like the user's full name.
+
+For more information on the `User` object, see the [`reference`](/docs/references/javascript/user/user).
+
+```tsx filename="user.tsx"
+import { useStore } from '@nanostores/react';
+import { $userStore } from '@clerk/astro/client';
+
+export default function User() {
+  const user = useStore($userStore);
+
+  if (user === undefined) {
+    // Handle loading state however you like
+    return null;
+  }
+
+  if (user) {
+    return <div>Hello {user.fullName}!</div>;
+  }
+
+  return <div>Not signed in</div>;
+}
+```
+
+### Update the current user data with the `$userStore` store
+
+The following example demonstrates how to use the `$userStore` Store to access the `user` object, which includes the `update` method for updating the user's data.
+
+For more information on the `update()` method, see the [`User`](/docs/references/javascript/user/user#update) reference.
+
+```tsx filename="user.tsx"
+import { useStore } from '@nanostores/react';
+import { $userStore } from '@clerk/astro/client';
+
+export default function User() {
+  const user = useStore($userStore);
+
+  if (user === undefined) {
+    // Handle loading state however you like
+    return null;
+  }
+
+  if (user === null) return null;
```

Comment:
`undefined` is equal to Clerk still loading, and `null` is no user? If so, let's actually add a section to this page calling that out. That's super useful info.

Reply:
I confirm that this is the correct logic

Reply:
Updated!